test/integration/http_integration.cc

```diff
@@ -1708,6 +1708,89 @@ void HttpIntegrationTest::testUpstreamProtocolError() {
   EXPECT_STREQ("503", response->headers().Status()->value().c_str());
 }
 
+void HttpIntegrationTest::testHeadersOnlyFilterEncoding() {
```

Comment:
I love these new tests. Any change of getting some tests with multiple intermediate filters? Possibly another encoder/decoder filter that just does pass through before getting to either the route on decode or the real encode write on the encoding side? I think this would also cover the encoder integration test ordering stuff that I had asked for in the other PR.

Reply:
Sure. Were you thinking just a dummy filter that returns continue for everything just to make sure nothing blows up, or are there explicit assertions you'd like? Like verifying that the filter is actually invoked?

Reply:
Mostly just making sure nothing blows up with multiple filters. Obviously, asserting that the callbacks get called would be great also.